src/app/ui.cljs

```diff
@@ -1,29 +1,43 @@
 (ns app.ui
   (:require [com.fulcrologic.fulcro.components :as comp :refer [defsc]]
-            [com.fulcrologic.fulcro.dom :as dom]))
+            [com.fulcrologic.fulcro.dom :as dom]
+            [com.fulcrologic.fulcro.routing.dynamic-routing :as dr]
+            [app.component.burger.build-controls :refer [toggle-modal]]
+            [app.component.burger.burger :refer [Burger]]))
 
-
-
-(defsc Backdrop [this {:ui/keys [modal?]}]
+(defsc Backdrop [this {:ui/keys [modal] :as props}]
   {}
-  (when modal?
-    (dom/div {:className "Backdrop" :onClick #(js/alert "backdrop")})))
+  (when modal
+    (dom/div {:className "Backdrop" :onClick
+                         #(comp/transact! this [(toggle-modal props)])})))
 
 (def ui-backdrop (comp/factory Backdrop))
 
 
-(defsc Modal [this {:ui/keys [modal?]
-                    :order/keys [burger] :as props}]
-  {:query [:ui/id :ui/modal? :order/burger]
+(defsc Modal [this {:ui/keys [modal burger] :as props}]
+  {:query [:ui/id :ui/modal {:ui/burger (comp/get-query Burger)}]
    :ident (fn [] [:singleton ::modal])
-   :initial-state {:ui/id 1
-                   :ui/modal? false}}
-  (js/console.log "Modal: " modal?)
-
-  (let [css (if modal? :.show :.noShow)]
-
-    (dom/div {:classes ["Modal" css]}
-      (ui-backdrop props))))
-
+   :initial-state (fn [_] {:ui/id 1
+                           :ui/modal false
+                           :ui/burger (comp/get-initial-state Burger)})}
+  (let [ingredients burger
+        css (if modal :.show :.noShow)
```

Comment:
Je hebt geen css gedefineerd voor `.show` en `.noShow`

Reply:
Die staan in de merged.css file bij alle andere CSS

Reply:
Dan kun je beter geen localized classnames gebruiken, als je nu de naam van het component wijzigt, de naam van de namespace wijzigt of het component naar een andere namespace verplaatst zal de css kapot gaan.